assets/components/src/color-picker/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * WordPress dependencies.
+ */
+import { ColorIndicator, ColorPalette } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import './style.scss';
+
+/**
+ * Color Picker.
+ */
+class ColorPicker extends Component {
+	state = {
+		color: '#ffffff',
+	};
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const { hasDefaultColors, label, suggestedColors, ...otherProps } = this.props;
+		const { color } = this.state;
```

Comment:
For components generally the value should be a prop, not state, so that the container can set and respond to changes. To accomplish this:

- Drop all the state management.
- Make `color` a `prop`.
- Make `onChange` a prop too, and for `ColorPalette`'s onChange use: `onChange={ onChange }`
- In Components Demo, add three new properties to state: `color1` `color2` `color3`
- Then for each `ColorPicker` component add attributes like this: `color={ color1 } onChange={ color => this.setState( { color1: color } ) }` (using `color2` and `color3` for the 2nd and 3rd)
- I'd also add defaultProps to the Component so it doesn't error out if there is no `onChange` set:

```
ColorPicker.defaultProps = {
       onChange: () => null,
       color: '#ffffff',
};
```

Reply:
Not sure if this is what you meant: https://github.com/Automattic/newspack-plugin/pull/431/commits/e0840c4789519c4042ab7ed02bd29c6a41d385c9 😅